File: 1700-Number-of-Students-Unable-to-Eat-Lunch.py
class Solution(object):
    def countStudents(self, students, sandwiches):
        n, k = len(students), sum(students)
        numZero, numOne = n-k, k

        for i in range(n):
            if sandwiches[i] == 0:
                if numZero == 0:
                    return n-i
                else:
                    numZero -= 1
            if sandwiches[i] == 1:
                if numOne == 0:
                    return n-i
                else:
                    numOne -= 1
        
        return 0

File: 1700-Number-of-Students-Unable-to-Eat-Lunch.js
var countStudents = function(students, sandwiches) {
    let n = students.length;
    let countOne = students.reduce((a, b) => a + b, 0);
    let countZero = n - countOne;
    
    for (let i = 0; i < n; i++) {
        if (sandwiches[i] === 0) {
            if (countZero === 0) {
                return n - i;
            } else {
                countZero--;
            }
        }
        if (sandwiches[i] === 1) {
            if (countOne === 0) {
                return n - i;
            } else {
                countOne--;
            }
        }        
    }
    
    return 0;
    
};